src/App.tsx

```diff
@@ -1,19 +1,187 @@
+/* eslint-disable no-console */
+/* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import './App.css';
 
+import { TodoList } from './components/TodoList/TodoList';
+import { PreparedTodo } from './type/preparedtodo';
+
+import todos from './api/todos';
 import users from './api/users';
+import { Todo } from './type/todo';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todosArr: PreparedTodo[],
+  userId: Todo['userId'],
+  title: Todo['title'],
+  userIdErrorMessage: string,
+  titleErrorMessage: string,
 };
 
+class App extends React.Component<{}, State> {
+  validForm = true;
+
+  state: State = {
+    todosArr: preparedTodos,
+    userId: 0,
+    title: '',
+    userIdErrorMessage: '',
+    titleErrorMessage: '',
+  };
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ userId: Number(event.target.value), userIdErrorMessage: '' });
```

Comment:
or use + instead of Number

```suggestion
    this.setState({ userId: Number(event.target.value), userIdErrorMessage: '' });
```

Reply:
```suggestion
    this.setState({ userId: +event.target.value, userIdErrorMessage: '' });
```

Reply:
ok